web/api/v1/api_test.go

```diff
@@ -1776,6 +1776,49 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, es storage.E
 			},
 			responseLen: 2,
 		},
+		// With a limit for the number of metadata per metric.
+		{
+			endpoint: api.metricMetadata,
+			query:    url.Values{"limit-per-metric": []string{"1"}},
```

Comment:
What about other test cases?

For example, you don't have a test case where you set the `metric` query parameter, which is especially important to test because the implementation code goes through a different path (aim for as close to 100% code coverage under unit tests).

Another example is you don't have a test where you have different metric metadata in a single target. All of your current tests are have the same metric metadata within each target.